src/Illuminate/Database/Query/Builder.php

```diff
@@ -2406,6 +2406,28 @@ public function doesntExist()
         return ! $this->exists();
     }
 
+    /**
+     * Execute the given callback if no rows exist for the current query.
+     *
+     * @param  \Closure $callback
+     * @return mixed
+     */
+    public function existsOr(Closure $callback)
+    {
+        return $this->exists() ? true : $callback();
```

Comment:
And return `$this->exists() ?: value($callback)`

I prefer `?:` over `? true :` but maybe that's just me.